.github/workflows/ui_tests.yml

```diff
@@ -110,9 +86,86 @@ jobs:
         env:
           EIP155_PRIVATE_KEY: ${{ secrets.EIP155_PRIVATE_KEY }}
           SOLANA_PRIVATE_KEY: ${{ secrets.SOLANA_PRIVATE_KEY }}
+          NODE_OPTIONS: '--max-old-space-size=4096' # Increase memory for webpack
         working-directory: ./apps/browser-extension/
         run: pnpm build
 
+      # Upload actual artifact in GitHub Actions
+      - name: Upload build artifacts (GitHub Actions)
+        uses: actions/upload-artifact@v4
+        with:
+          name: ui-build-output
+          path: |
+            dist/
+            .next/
+            turbo/
+            packages/*/dist/
+            apps/*/dist/
+            apps/*/.next/
+            apps/*/out/
+            .turbo/
+          retention-days: 1
+
+      # Upload actual artifact in GitHub Actions
+      - name: Upload node_modules (GitHub Actions)
+        uses: actions/upload-artifact@v4
+        with:
+          name: ui-node-modules
+          path: node_modules
+          retention-days: 1
+
+  ui_tests:
+    name: 'Playwright Tests'
+    needs: build_lab_and_extension # Depend on the build job
```

Comment:
doesn't it require the upload as well?

Reply:
the upload is part of that? Not sure I understand?